packages/rpc-transport/src/transports/websocket/websocket-connection.ts

```diff
@@ -29,7 +35,21 @@ export async function createWebSocketConnection({
     return new Promise((resolve, reject) => {
         signal.addEventListener('abort', handleAbort, { once: true });
         const iteratorState: Map<IteratorKey, IteratorState> = new Map();
+        function errorAndClearAllIteratorStates(reason: unknown) {
+            const errorCallbacks = [...iteratorState.values()]
+                .filter((state): state is Extract<IteratorState, { __hasPolled: true }> => state.__hasPolled)
+                .map(({ onError }) => onError);
+            iteratorState.clear();
+            errorCallbacks.forEach(cb => {
+                try {
+                    cb(reason);
+                } catch {
+                    /* empty */
+                }
+            });
+        }
         function handleAbort() {
+            errorAndClearAllIteratorStates(EXPLICIT_ABORT_TOKEN);
```

Comment:
This is the material change; blow away all the iterators immediately upon having called `abort()`

Reply:
with `AsyncIterator`s, that means I wouldn't lose any messages that have already arrived, right? I'd just lose the ability to poll for next?

Reply:
That's right. If you've polled and they've resolved, they're in your queue, but if you're still waiting for a message to arrive and you abort the transport, we'll return from your poll immediately.